Java 11: DEFAULT_JAXWSPROVIDER seems wrong

I use Java 11, and `javax.xml.ws.spi.Provider` (2.3.2) wants to resolve `com.sun.xml.internal.ws.spi.ProviderImpl` which is not provided by `com.sun.xml.ws.jaxws-rt` (2.3.2), where this file is called `com.sun.xml.ws.spi.ProviderImpl`.

What is the right combination to use here??

i'm stuck on this trying to upgrade an application to Java 11.
In my local developer machine works as expected, but in other environments i get a ClassNotFound.

Not being able to solve the issue following the stackoverflow link provided above.

```
javax.xml.ws.WebServiceException: Provider com.sun.xml.internal.ws.spi.ProviderImpl not found
...
Caused by: java.lang.ClassNotFoundException: com.sun.xml.internal.ws.spi.ProviderImpl
```

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -1,6 +1,6 @@
 <!--
 
-    Copyright (c) 1997, 2020 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 1997, 2021 Oracle and/or its affiliates. All rights reserved.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Distribution License v. 1.0, which is available at
@@ -106,7 +106,7 @@
         <spotbugs.exclude>${project.basedir}/exclude.xml</spotbugs.exclude>
         <spotbugs.skip>false</spotbugs.skip>
         <spotbugs.threshold>Low</spotbugs.threshold>
-        <spotbugs.version>4.0.4</spotbugs.version>
+        <spotbugs.version>4.2.0</spotbugs.version>
 
         <release.spec.feedback>jaxws-dev@eclipse.org</release.spec.feedback>
 
@@ -120,8 +120,8 @@
         <legal.doc.source>${project.basedir}/..</legal.doc.source>
 
         <jws-api.version>3.0.0</jws-api.version>
-        <soap-api.version>2.0.0</soap-api.version>
-        <xml.bind-api.version>3.0.0</xml.bind-api.version>
+        <soap-api.version>2.0.1</soap-api.version>
+        <xml.bind-api.version>3.0.1</xml.bind-api.version>
     </properties>
 
     <build>
@@ -175,7 +175,7 @@
                 <plugin>
                     <groupId>org.glassfish.copyright</groupId>
                     <artifactId>glassfish-copyright-maven-plugin</artifactId>
-                    <version>2.3</version>
+                    <version>2.4</version>
                 </plugin>
                 <plugin>
                     <groupId>org.glassfish.build</groupId>
@@ -238,6 +238,7 @@
                     <update>${copyright.update}</update>
                     <!-- check that year is correct -->
                     <ignoreYear>${copyright.ignoreyear}</ignoreYear>
+                    <quiet>false</quiet>
                 </configuration>
                 <executions>
                     <execution>
@@ -271,6 +272,18 @@
                 <artifactId>build-helper-maven-plugin</artifactId>
                 <executions>
                     <execution>
+                        <id>currentyear-property</id>
+                        <goals>
+                            <goal>timestamp-property</goal>
+                        </goals>
+                        <phase>validate</phase>
+                        <configuration>
+                            <name>current.year</name>
+                            <locale>en,US</locale>
+                            <pattern>yyyy</pattern>
+                        </configuration>
+                    </execution>
+                    <execution>
                         <id>add-legal-resource</id>
                         <phase>generate-resources</phase>
                         <goals>
@@ -295,8 +308,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>9</source>
-                    <target>9</target>
+                    <release>9</release>
                     <compilerArgs>
                         <arg>-Xlint:all</arg>
                     </compilerArgs>
@@ -420,7 +432,7 @@
                     <header><![CDATA[Jakarta XML Web Services<br>v${spec.version}]]></header>
                     <bottom><![CDATA[
 Comments to: <a href="mailto:${release.spec.feedback}">${release.spec.feedback}</a>.<br>
-Copyright &#169; 2019, 2020 Eclipse Foundation. All rights reserved.<br>
+Copyright &#169; 2019, ${current.year} Eclipse Foundation. All rights reserved.<br>
 Use is subject to <a href="{@docRoot}/doc-files/speclicense.html" target="_top">license terms</a>.]]>
                     </bottom>
                 </configuration>

--- a/api/src/main/java/jakarta/xml/ws/spi/FactoryFinder.java
+++ b/api/src/main/java/jakarta/xml/ws/spi/FactoryFinder.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2005, 2020 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2005, 2021 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Distribution License v. 1.0, which is available at
@@ -71,7 +71,10 @@ class FactoryFinder {
 
         // handling Glassfish (platform specific default)
         if (isOsgi()) {
-            return (T) lookupUsingOSGiServiceLoader(factoryId);
+            provider = (T) lookupUsingOSGiServiceLoader(factoryId);
+            if (provider != null) {
+                return provider;
+            }
         }
 
         if (fallbackClassName == null) {

--- a/api/src/main/java/jakarta/xml/ws/spi/Provider.java
+++ b/api/src/main/java/jakarta/xml/ws/spi/Provider.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2005, 2020 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2005, 2021 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Distribution License v. 1.0, which is available at
@@ -30,9 +30,7 @@ public abstract class Provider {
      * A constant representing the name of the default
      * {@code Provider} implementation class.
      **/
-    // Using two strings so that package renaming doesn't change it
-    private static final String DEFAULT_JAXWSPROVIDER =
-            "com.sun"+".xml.internal.ws.spi.ProviderImpl";
+    private static final String DEFAULT_JAXWSPROVIDER = "com.sun.xml.ws.spi.ProviderImpl";
 
     /**
      * Creates a new instance of Provider

